diff --git a/tests/test_rio_warp.py b/tests/test_rio_warp.py
index 317f22010..135d8a000 100644
--- a/tests/test_rio_warp.py
+++ b/tests/test_rio_warp.py
@@ -2,6 +2,7 @@
 
 
 import os
+import re
 
 import affine
 import numpy as np
@@ -12,6 +13,11 @@
 from rasterio.rio import warp
 from rasterio.rio.main import main_group
 
+# Make a gdal version tuple.
+version_name = rasterio.gdal_version()
+version_name = re.split(r"[a-z]", version_name)[0]
+gdal_version_info = tuple(int(x) for x in version_name.split("."))
+
 
 def test_dst_crs_error(runner, tmpdir):
     """Invalid JSON is a bad parameter."""
@@ -590,6 +596,11 @@ def test_unrotate(runner, tmp_path):
         assert src.transform.d == 0.0
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="Warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 @pytest.mark.parametrize(
     "wotopt", ["--to", "--transformer-option", "--wo", "--warper-option"]
 )
diff --git a/tests/test_warp.py b/tests/test_warp.py
index e2c183436..3f49948ff 100644
--- a/tests/test_warp.py
+++ b/tests/test_warp.py
@@ -1339,6 +1339,11 @@ def test_reproject_resampling(path_rgb_byte_tif, method):
     assert np.count_nonzero(out) in expected[method]
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 @pytest.mark.parametrize(
     "test3d,count_nonzero",
     [
@@ -1404,6 +1409,11 @@ def dtype(self):
     assert np.count_nonzero(out.data[out.data != 99]) == count_nonzero
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 @pytest.mark.parametrize(
     "test3d,count_nonzero",
     [
@@ -1515,6 +1525,11 @@ def test_reproject_to_masked_output(path_rgb_byte_tif):
     assert isinstance(out, np.ma.MaskedArray)
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 @pytest.mark.parametrize(
     "test3d,count_nonzero",
     [
@@ -2397,6 +2412,11 @@ def test_rpcs_non_epsg4326():
             )
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 def test_coordinate_pipeline(tmp_path):
     """Transformer COORDINATE_OPERATION option is activated."""
     pipeline = "proj=pipeline step inv proj=utm zone=11 ellps=clrk66 step proj=unitconvert xy_in=rad xy_out=deg step proj=axisswap order=2,1"
@@ -2432,6 +2452,11 @@ def test_coordinate_pipeline(tmp_path):
             assert dst.checksum(1) == 4705
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 def test_geoloc_warp_dataset(data, tmp_path):
     """Warp a dataset using external geolocation arrays."""
     filename = str(data.join("RGB.byte.tif"))
@@ -2476,6 +2501,11 @@ def test_geoloc_warp_dataset(data, tmp_path):
     assert np.count_nonzero(out) in [464910]
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 # Before GDAL 3.5.2 geoloc array files aren't recognized and this error
 # would be seen from the following tests:
 #
@@ -2512,6 +2542,11 @@ def test_geoloc_warp_array(path_rgb_byte_tif):
     assert np.count_nonzero(output[0]) in [464910]
 
 
+@pytest.mark.xfail(
+    gdal_version_info >= (3, 12, 2),
+    reason="warper implementation has changed. See https://github.com/rasterio/rasterio/issues/3517.",
+    raises=AssertionError,
+)
 def test_geoloc_warp_array_subsampled(path_rgb_byte_tif):
     """Warp an array using subsampled external geolocation arrays."""
     with rasterio.open(path_rgb_byte_tif) as src:
